### DISCUSSION TITLE
v3.2.0

### DISCUSSION THREAD

**Discussion by danharrin** [filamentphp/filament#10807] [General]
## What's Changed
* Support radio buttons and button group by @zepfietje in https://github.com/filamentphp/filament/pull/9860
* Password reveal by @Stichoza and @danharrin in https://github.com/filamentphp/filament/pull/10326
* Add HasDescription for enum and for Radio by @fouteox in https://github.com/filamentphp/filament/pull/10530
* feature: Apply table filters by @danharrin in https://github.com/filamentphp/filament/pull/10531
* feature: Unsaved changes alert by @danharrin in https://github.com/filamentphp/filament/pull/10516
* Add forms split by @zepfietje in https://github.com/filamentphp/filament/pull/9998
* Infolists: Add `KeyValueEntry` component by @juliomotol in https://github.com/filamentphp/filament/pull/10557
* Rework toggle buttons by @zepfietje in https://github.com/filamentphp/filament/pull/10623
* feature: Column groups by @danharrin in https://github.com/filamentphp/filament/pull/10744
* Add German translation for text column actions collapse/expand list by @tobias-grasse in https://github.com/filamentphp/filament/pull/10781
* Uzbek language translations added by @encoderuz in https://github.com/filamentphp/filament/pull/10784
* feature: Export action by @danharrin in https://github.com/filamentphp/filament/pull/10793
* feature: Filter layout by @danharrin in https://github.com/filamentphp/filament/pull/10794
* feature: Form component action testing by @danharrin in https://github.com/filamentphp/filament/pull/10796
* feature: Infolist action testing by @danharrin in https://github.com/filamentphp/filament/pull/10797
* Addition of 'whereCustomProperties' to Spatie media library file upload by @chrispage1 in https://github.com/filamentphp/filament/pull/10795
* Add panel css selector by @binaryfire in https://github.com/filamentphp/filament/pull/10799
* fix: Do not save field relationships when components are disabled by @danharrin in https://github.com/filamentphp/filament/pull/10804
* fix: afterStateUpdated() not executing for batched requests by @danharrin in https://github.com/filamentphp/filament/pull/10805

## New Contributors
* @tobias-grasse made their first contribution in https://github.com/filamentphp/filament/pull/10781
* @encoderuz made their first contribution in https://github.com/filamentphp/filament/pull/10784
* @chrispage1 made their first contribution in https://github.com/filamentphp/filament/pull/10795

**Full Changelog**: https://github.com/filamentphp/filament/compare/v3.1.46...v3.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/filamentphp/filament/releases/tag/v3.2.0'>v3.2.0</a>.</em>

**Comment by marcosmarcolin** [comment]
@danharrin Hi, I just upgraded to local testing and had a problem with https://github.com/filamentphp/filament/pull/10516.

I'm using `->spa()`, but at no point have I activated `->unsavedChangesAlerts()` as it says here: https://filamentphp.com/docs/3.x/panels/configuration#unsaved-changes-alerts

Exception: `Unsaved changes alerts are not supported in SPA mode.`

Of course if I remove `->spa()`, it works normally.

But if I don't use `->unsavedChangesAlerts()`, `->spa()` should work, right?

(This seems like a BC.)

**Reply by danharrin** [reply]
Good catch! Released a patch in v3.2.1
